### PR TITLE
Worker monitoring

### PR DESCRIPTION
**Pull request by jrmbrgs**

### CODE REVIEW

File: Library/Monitoring/Service.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+     *
+     * @param string $workerCode the worker's code to register
+     *
+     * @return void
+     */
+    public function workerBeat(string $workerCode)
+    {
+        $this->di->get('logr')->debug(json_encode(func_get_args()));
+        $workerHash = [
+            'heartBeatAt' => Date('Y-m-d H:i:s')
```
Comment (mventuravoyageprive) [inline Library/Monitoring/Service.php]: date function should be in milluscule

File: Library/Monitoring/Service.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+     * @return void
+     */
+    public function workerStarted(string $workerCode, int $pid)
+    {
+        $this->di->get('logr')->debug(json_encode(func_get_args()));
+        $workerHash = [
+            'status' => Core\AbstractWorker::STATUS_STARTED,
+            'runingOn' => php_uname("n"),
+            'pid' => $pid,
+            'startedAt' => Date('Y-m-d H:i:s'),
```
Comment (mventuravoyageprive) [inline Library/Monitoring/Service.php]: date function should be in milluscule

File: Library/Monitoring/Service.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            'heartBeatAt' => Date('Y-m-d H:i:s')
```
Comment (mventuravoyageprive) [inline Library/Monitoring/Service.php]: date function should be in milluscule

File: Library/Monitoring/Service.php [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+     *
+     * @param string $workerCode the worker's code to register
+     *
+     * @return void
+     */
+    public function workerHeartBeats(string $workerCode)
+    {
+        $this->di->get('logr')->debug(json_encode(func_get_args()));
+        $workerHash = [
+            'heartBeatAt' => Date('Y-m-d H:i:s')
```
Comment (mventuravoyageprive) [inline Library/Monitoring/Service.php]: date function should be in milluscule

File: Library/Monitoring/Service.php [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+     * @param string $workerCode the worker's code to register
+     * @param int    $exitCode   the worker's exit code
+     *
+     * @return void
+     */
+    public function workerExited(string $workerCode, int $exitCode = 0)
+    {
+        $this->di->get('logr')->debug(json_encode(func_get_args()));
+        $workerHash = [
+            'exitedAt' => Date('Y-m-d H:i:s'),
```
Comment (mventuravoyageprive) [inline Library/Monitoring/Service.php]: date function should be in milluscule

File: Library/Monitoring/Service.php [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+    /**
+     * Registers a worker into the monitoring sys
+     *
+     * @param string $workerCode the worker's code to register
+     * @param int    $pid        the worker's pid
+     *
+     * @return void
+     */
+    public function workerStarted(string $workerCode, int $pid)
```
Comment (mventuravoyageprive) [inline Library/Monitoring/Service.php]: Change order in function name startWorker

**Review by mventuravoyageprive — COMMENTED** [review]
Replace logr

File: Library/Monitoring/Service.php (hunk 1 above)
Comment (jrmbrgs) [reply Library/Monitoring/Service.php]: done

File: Library/Monitoring/Service.php (hunk 6 above)
Comment (jrmbrgs) [reply Library/Monitoring/Service.php]: infact, here we monitor a past action "worker started" 
if we define a "startWorker" func it's supposed to "start a worker"
Maybe you want the func to start w/ a verb. why not `logWorkerStarted` ?

**Pull request merged**